test/integration/gen_tests.py

```diff
@@ -250,6 +250,32 @@ def gen_target(output_target_name, test_configs):
         add_footer(f)
 
 
+def gen_target_shlib(output_target_name, test_configs):
+    with open(output_target_name, "w") as f:
+        headers = set()
+        for config in test_configs:
+            headers.update(config.get("includes_shlib", []))
```

Comment:
Shall we just use the same "includes" for the shared lib? It'd simplify writing the test configs and the worst that'll happen is that we'll get unnecessary includes

Reply:
Otherwise, "includes_shlib" should be documented too